### PR TITLE
Merge feature/non-cdn-update to master

### PR DESCRIPTION
**Pull request by BengangY**
Merge `feature/non-cdn-update` to `master`.
Resolved conflicts:
    ocaml/tests/dune
    quality-gate.sh

### CODE REVIEW

**Comment by minglumlu** [comment]
I think we need the auto-generated changes in `datamodel_lifecycle.ml`. And bumping up minor `vsn` together with the `last_known_schema_hash`.

**Comment by BengangY** [comment]
> I think we need the auto-generated changes in `datamodel_lifecycle.ml`. And bumping up minor `vsn` together with the `last_known_schema_hash`.

Will update it on another PR https://github.com/xapi-project/xen-api/pull/5927.

**Review by minglumlu — APPROVED** [review]

File: ocaml/tests/test_tar_ext.ml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+open Helpers
+open Tar_ext
+open Bos
+
+let ( // ) = Filename.concat
+
+let gen_test_file_script = "test_data" // "gen_tar_ext_test_file.sh"
+
+let max_size_limit = 2000000L
```
Comment (lindig) [inline ocaml/tests/test_tar_ext.ml:23]: This would be more readable as `2_000_000L` or (`2 * 1024 * 1024 |> Int64.of_int`)

**Review by lindig — APPROVED** [review]

File: ocaml/tests/test_tar_ext.ml (hunk 1 above)
Comment (gangj) [inline ocaml/tests/test_tar_ext.ml:23]: There is no link between the limit here and the shell script which generates a file bigger than this size, I think we should at least add some comments to make the relationship clear, to make this test and the shell script easy to maintain later.

**Review by gangj — APPROVED** [review]

**Pull request merged**